### ISSUE TITLE
Frame drop with 8K av1 or hevc video files with HW acceleration on, though CPU/Disk/GPU are not saturated yet.

### ISSUE THREAD

**Issue by jerryrt** [mpv-player/mpv#11570]
Any suggestion on how to investigate the issue will be much appreciated.

BTW, I can confirm the GPU is not the bottleneck here, at least for decoding(AV1, HEVC are both tested). I have build FFMPEG with NVDEC support from source code, and benchmarked the GPU for decoding speed,  like this:

> ffmpeg -hwaccel cuvid -i Switzerland\ in\ 8K\ ULTRA\ HD\ HDR\ -\ Heaven\ of\ Earth\ \(60\ FPS\)\ \[linlz7-Pnvw\].mkv -f null

The sample file is downloaded from [YouTube](https://www.youtube.com/watch?v=linlz7-Pnvw) using yt-dlp.

I can got around 120fps decoding speed, and GPU monitor tools can confirm that GPU decoding utilization is reaching 100%.

But with mpv, CPU is mostly idle, and the same metric for GPU decoding is around 40%-50%, with constantly frame dropping.

Provide following Information:
- mpv version

> $mpv --version
> mpv 0.34.1 Copyright © 2000-2021 mpv/MPlayer/mplayer2 projects
>  built on UNKNOWN
> FFmpeg library versions:
>    libavutil       56.70.100
>    libavcodec      58.134.100
>    libavformat     58.76.100
>    libswscale      5.9.100
>    libavfilter     7.110.100
>    libswresample   3.9.100
> FFmpeg version: 4.4.2-0ubuntu0.22.04.1

- Linux Distribution and Version

> $ uname -a
> Linux smallpot 5.19.0-38-generic #39~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Mar 17 21:16:15 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

- Source of the mpv binary

Ubuntu official repository

- If known which version of mpv introduced the problem

Not aware of.

- Window Manager and version

> $ update-alternatives --list x-window-manager
> /usr/bin/i3
> /usr/bin/openbox
> 
> $ i3 --version
> i3 version 4.20.1 (2021-11-03) © 2009 Michael Stapelberg and contributors

- GPU driver and version

> $ nvidia-smi
> Tue Apr 11 16:41:36 2023       
> +-----------------------------------------------------------------------------+
> | NVIDIA-SMI 525.105.17   Driver Version: 525.105.17   CUDA Version: 12.0     |
> ...

- Possible screenshot or video of visual glitches

None

### Reproduction steps

HWDEC enabled in config file, and logging output confirms this.

> $ cat ~/.config/mpv/mpv.conf 
> vo=gpu
> hwdec=auto

### Expected behavior

8K playback should be smooth.

### Actual behavior

Around 1/3-1/2 frames are dropped.

### Log file

https://gist.github.com/jerryrt/f8786025404c26f8141516cc77fd7b37

### Sample files

The sample file is downloaded from [YouTube](https://www.youtube.com/watch?v=linlz7-Pnvw) using yt-dlp.

**Comment by jerryrt** [comment]
> If `video-unscaled=no`, the frame rate will drop sharply and it can be observed that the GPU usage does not reach 100%, maybe the bottleneck is in the VRAM bandwidth?

Yes, from my experience, I doubt some extra copy forth and back in VRAM even between RAM is the bottleneck. But most of my experience is in mobile development, I am all ears here.

**Comment by jerryrt** [comment]
![Screenshot_01](https://user-images.githubusercontent.com/1451147/231178736-8f7da741-8662-4bd8-80c6-84706bedb471.png)

Changed to another machine with more powerful GPU(3080Ti), frame drop improved a lot, around 2-3 frames every second, at least not lagging visually. However, GPU decoder still not saturated, at least according to metris reported by nvtop(above image).

VRAM usage still the same, and no pressure on the CPU.

**Comment by jerryrt** [comment]
> pause = yes
> osd-msg1 = "FPS: ${estimated-display-fps}"
> opengl-swapinterval = 0
> opengl-dwmflush = no
> vulkan-swap-mode = immediate
> framedrop = no
> hidpi-window-scale = no
> video-unscaled = yes
> sub = no
> audio = no
> keep-open = always
> video-sync = display-desync
> display-fps=60

Confirmed, with the above options enabled, the FPS output matched numbers from ffmpeg benchmark mode, which is around 120fps.

**Comment by philipl** [comment]
You should try with `vo=gpu-next` and `gpu-api=vulkan`. That will be more performant.

I can't give you a like-for-like comparison, but at least on a 4090, it can display without dropping frames.

**Comment by philipl** [comment]
But even on the 4090, if I use a too-fancy scaling algorithm, it will drop a couple of frames each second.

**Comment by philipl** [comment]
This doesn't drop frames for me.

```
mpv Switzerland\ in\ 8K\ ULTRA\ HD\ HDR\ -\ Heaven\ of\ Earth\ \(60\ FPS\)\ \[linlz7-Pnvw\].webm --no-config --hwdec=cuda --vo=gpu-next --gpu-api=vulkan --profile=gpu-hq
```

**Comment by jerryrt** [comment]
Hi all, thanks for all the suggestions.

Finally pinpoint the exact configure option that caused the framedrop on my RTX3060 system, by one by one commenting out configure options given by @arm64-v9a 

After enable this option, frame drop reduced to 2-3 frame/second, acceptable for daily usage. 

> opengl-swapinterval = 0

Hopes this help anyone with similar issues on Linux system.

> You should try with `vo=gpu-next` and `gpu-api=vulkan`. That will be more performant.
> 
> I can't give you a like-for-like comparison, but at least on a 4090, it can display without dropping frames.

At least the official package from Ubuntu Jammy's version is too old for this output options, when time avaiable for compiling my own mpv, I will give it a try to check if it improves.

**Comment by jerryrt** [comment]
Found the work around with vo=gpu backend.
